docs/root/start/sandboxes/single-page-app.rst

```diff
@@ -0,0 +1,648 @@
+.. _install_sandboxes_single_page_app:
+
+Single page React app (with OAuth)
+==================================
+
+.. sidebar:: Requirements
+
+   .. include:: _include/docker-env-setup-link.rst
+
+   :ref:`curl <start_sandboxes_setup_curl>`
+        Used to make HTTP requests.
+
+   :ref:`envsubst <start_sandboxes_setup_envsubst>`
+        Used to interpolate environment vars in templates.
+
+   :ref:`jq <start_sandboxes_setup_jq>`
+        Used to parse JSON.
+
+   :ref:`mkpasswd <start_sandboxes_setup_mkpasswd>`
+        Used to generate a ~random HMAC token.
+
+This sandbox provides an example of building and developing a single page app with Envoy.
+
+The sandbox covers a number of Envoy's features, including:
+
+- :ref:`direct_response <config_network_filters_direct_response>`
+- :ref:`OAuth <config_http_filters_oauth>`
+- Dynamic xDS filesystem updates
+- Websocket proxy
+- Gzip :ref:`compression <config_http_filters_compressor>`
+- TLS/SNI up/downstream connection/termination
+- Path/host rewrites
+
+The app is built with `React <react_>`__ using `Vite <vite_>`__ and demonstrates OAuth authentication using
+Envoy's :ref:`OAuth filter <config_http_filters_oauth>`.
+
+This covers a scenario where we want OAuth to both authenticate the user and provide credentials
+for further API interactions.
+
+This is enabled by setting the OAuth configuration
+:ref:`forward_bearer_token <envoy_v3_api_field_extensions.filters.http.oauth2.v3.OAuth2Config.forward_bearer_token>`
+to ``true``
+
+.. literalinclude:: _include/single-page-app/envoy.yml
+   :language: yaml
+   :lines: 36-40
+   :linenos:
+   :lineno-start: 36
+   :emphasize-lines: 3
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+.. warning::
+   Setting
+   :ref:`forward_bearer_token <envoy_v3_api_field_extensions.filters.http.oauth2.v3.OAuth2Config.forward_bearer_token>`
+   means the provided access token will be forwarded to upstreams proxied by Envoy unless explicitly excluded.
+
+   This can be avoided by disabling the OAuth2 filter with
+   :ref:`per-route configuration <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpFilter.disabled>`.
+
+A dummy "Myhub" backend is provided with a minimal OAuth provider and API for use in the example.
+
+Setup is provided to :ref:`build and update the app for production use <install_sandboxes_single_page_app_step_production_build>`,
```

Comment:
its hard not to call it `production` becase that is what js/node calls it when you build the assets rather than serve them

eg - it reads `.env.production` in that circumstance